packages/astro/src/core/render/route-cache.ts

```diff
@@ -58,6 +58,21 @@ export async function callGetStaticPaths({
 		// Q: Why the cast?
 		// A: So users downstream can have nicer typings, we have to make some sacrifice in our internal typings, which necessitate a cast here
 		paginate: generatePaginateFunction(route) as PaginateFunction,
+		pathToParam(path: string) {
```

Comment:
No intention to blow up the scope, but do you think users might expect this to do more normalization than it does? Like remove invalid characters or something

Reply:
I wouldn't be surprised but I can't anticipate what those expectations are. Should we not do this until we have a better idea?

Reply:
I also expected that. I think we should call this function `unstable_pathToParam` and see how it's received, and we figure out what it's meant for (should it encode stuff? I mean, `/folder name` is a valid path but not a valid param because the space should be encoded. 

Having it called `unstable_` (or `experimental_*`) would allow us to opt in to our experimentation contract and break stuff whenever we see fit.